### ISSUE TITLE
ImportError: No module named Keccak

### ISSUE THREAD

**Issue by msg768** [bitcoin-abe/bitcoin-abe#253]
I'm not sure why I keep getting this. I have installed pysha3. Is there anything else I need to do?

**Comment by Lvl4Sword** [comment]
I don't believe pysha3 has a 'Keccak' to import ( looks like it's just monkeypatching hashlib )

From https://pypi.python.org/pypi/pysha3 it specifically says this too: "The sha3 module monkey patches the hashlib module ."

Did you mean instead to do
`from Crypto.Hash import keccak` ?

**Comment by dermoth** [comment]
it's KeccakChain not Keccak.

Here's for example the __non-functional__ SmartCash.py module I added to Abe... It still needs more work to make it fully load blocks though. 

    from .KeccakChain import KeccakChain
    from .. import util
    
    class Smartcash(KeccakChain):
        """
        Smartcash blockchain (https://smartcash.cc/)
        """
        def __init__(chain, **kwargs):
            chain.name = 'Smartcash'
            chain.code3 = 'SMART'
            chain.address_version = '\x3f'
            chain.script_addr_vers = '\x12'
            chain.magic = '\x5c\xa1\xab\x1e'
            KeccakChain.__init__(chain, **kwargs)

The `sha3` module is imported by `Abe/util.py` only if needed (technically, always since Abe doesn't run yet on Python3) then hashlib is indeed monkey-patched to include sha3 methods.
